pkg/charts/linkerd2/values.go

```diff
@@ -31,7 +31,7 @@ type (
 		ControllerUID                int64                  `json:"controllerUID"`
 		EnableH2Upgrade              bool                   `json:"enableH2Upgrade"`
 		EnablePodAntiAffinity        bool                   `json:"enablePodAntiAffinity"`
-		NodeAffinity                 map[string]string      `json:"nodeAffinity"`
+		NodeAffinity                 map[string]interface{} `json:"nodeAffinity"`
```

Comment:
Is there a reason to avoid the [upstream types](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2692-L2721) instead of using `interface{}`?

Reply:
Not a good reason. I thought for a quick fix I'd just be consistent with the way we're doing things there. Better to do it now than delay it though, so I've made the change! Good suggestion :) 

(I also trust explicit types more than interface{} so I'm liking the change this way)

Reply:
I think the choice of type here depends on how much we want to take responsibility for this field.  By using an explicit type, the Linkerd CLI will throw an error if a user provides an invalid NodeAffinity.  Alternatively, if we leave this untyped (interface{}), we'll just pass through whatever the user provides and let Kubernetes handle validation and let Kubernetes reject the resource if it's invalid.

Passing responsibility off to Kubernetes has the advantage that we don't tie ourselves to a specific schema for NodeAffinity (similarly for other types of other fields where we use interface{}) and allows the Linkerd CLI to more easily work with multiple versions of Kubernetes if the internal structure of these types changes.

On the other hand, the kubernetes APIs are probably stable enough and use deprecation cycles.  So this probably isn't a big deal.  Just some food for thought.

Reply:
@adleong thank you for the explanation! I was a bit naive in my initial assumptions and didn't consider the implications this would have on keeping up with different k8s versions.

Based on your explanation, I'll revert the last commit and leave `nodeAffinity` untyped. While I think the suggestion is good, perhaps we shouldn't be so hasty in making the change. I can see that it requires a bit more thought and conversation. I've raised https://github.com/linkerd/linkerd2/issues/9006 to change the types.